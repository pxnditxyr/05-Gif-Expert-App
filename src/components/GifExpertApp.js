import React, { useState } from 'react';
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';

export const GifExpertApp = () => {

	// const categories = [ "Naruto", "Bleach", "One Piece" ];
	//const [ categories, setCategories ] = useState( ["Naruto", "Bleach", "One Piece", "KissxSis"] );
	const [ categories, setCategories ] = useState( ["Naruto"] );


	// const handleAdd = () => setCategories( [...categories, "KissxSis"] )

	return (

		<>
			<h2> GifExpertApp XD </h2>
			<AddCategory setCategories={ setCategories } />
			<hr />
			{/* <button onClick={ handleAdd }> Add </button> */}
			<ol>
				{
					categories.map( category =>
						<GifGrid
							key={ category }
							category={ category }
						/>
						//{/* <li key={ category }> { category } </li> */}
					)
				}
			</ol>
		</>
	);
};
