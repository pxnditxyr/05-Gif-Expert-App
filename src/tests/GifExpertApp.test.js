import { GifExpertApp } from '../GifExpertApp';
import { shallow } from 'enzyme';

describe( "Test's on GifExpertApp component", () => {
	test( "Should show the component", () => {
		const wrapper = shallow( <GifExpertApp /> );
		expect( wrapper ).toMatchSnapshot();
	});

	test( "Should show a list of categories", () => {
		const categories = [ "Naruto", "Ichigo", "Luffy" ];
		const wrapper = shallow( <GifExpertApp defaultCategories={ categories }/> );

		expect( wrapper ).toMatchSnapshot();

		expect( wrapper.find( 'GifGrid' ).length ).toBe( categories.length );

	});
});
