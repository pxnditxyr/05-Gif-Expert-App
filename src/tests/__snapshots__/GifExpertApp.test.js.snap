// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test's on GifExpertApp component Should show a list of categories 1`] = `
<Fragment>
  <h2>
     GifExpertApp XD 
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Naruto"
      key="Naruto"
    />
    <GifGrid
      category="Ichigo"
      key="Ichigo"
    />
    <GifGrid
      category="Luffy"
      key="Luffy"
    />
  </ol>
</Fragment>
`;

exports[`Test's on GifExpertApp component Should show the component 1`] = `
<Fragment>
  <h2>
     GifExpertApp XD 
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
